src/main/java/com/team4/testingsystem/repositories/AmazonS3Repository.java

```diff
@@ -0,0 +1,75 @@
+package com.team4.testingsystem.repositories;
+
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectInputStream;
+import com.amazonaws.util.IOUtils;
+import com.team4.testingsystem.exceptions.FileDeletingFailedException;
+import com.team4.testingsystem.exceptions.FileLoadingFailedException;
+import com.team4.testingsystem.exceptions.FileSavingFailedException;
+import org.apache.commons.io.FileUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Repository;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+@Profile("release")
+@Repository
+public class AmazonS3Repository implements FilesRepository {
+    private final AmazonS3 amazonS3;
+
+    @Value("${cloud.s3.bucket-name}")
+    private String bucketName;
+
+    @Autowired
+    public AmazonS3Repository(AmazonS3 amazonS3) {
+        this.amazonS3 = amazonS3;
+    }
+
+    @Override
+    public void save(String fileName, Resource file) throws FileSavingFailedException {
+        try {
+            Path tempFilePath = createTempFile(fileName);
+            File tempFile = new File(tempFilePath.toString());
+
+            FileUtils.copyInputStreamToFile(file.getInputStream(), tempFile);
```

Comment:
There's another option, `putObject` can be used with `InputStream` directly, but then `ObjectMetadata` should be passed (like content-length and so on).
It seems better, I can search how to implement this option.

Reply:
I've ask the frontend guys if they always can receive such info.
Anyway, we can try to use the solution for now and update it in case something will go wrong

Reply:
They are not sure if the can it. Do, we can't use it

Reply:
Why we can't?
Documentations says that we must specify only `content-length`, that we can get from `MultipartFile`.
So, maybe we can pass content length as a parameter?

Reply:
put object requires content-length that frontend may not be able to provide.
Buffering all the files seems weird and could cause OOM exceptions.

The multipart upload will force us to make our own buffer and send the chunks. If the file will be smaller then the buffer - we can just post it as well